File: backend/src/main/java/com/tasklion/backend/features/task/model/CommonTaskRequestModel.java
package com.tasklion.backend.features.task.model;

import com.tasklion.backend.features.taskerService.serviceArea.model.ServiceAreaModel;
import com.tasklion.backend.features.taskerService.serviceCategory.model.ServiceCategoryModel;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalTime;

@Data
@Builder
public class CommonTaskRequestModel {

    private ServiceCategoryModel serviceCategory;
    private LocalDate date;
    private LocalTime startTime;
    private LocalTime endTime;
    private ServiceAreaModel serviceArea;
    private String remarks;

}

File: frontend/src/app/modules/task/models/common-task-request.model.ts
import {ServiceCategoryModel} from "../../tasker-service/models/service-category.model";
import {ServiceAreaModel} from "../../tasker-service/service-area/models/service-area.model";

export interface CommonTaskRequestModel {
  serviceCategory: ServiceCategoryModel;
  date: string;
  startTime: string;
  endTime: string;
  serviceArea: ServiceAreaModel
  remarks?: string;
}
